SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp

```diff
@@ -464,7 +464,11 @@ void USpatialWorkerConnection::OnConnectionFailure()
 			const FString ErrorMessage(UTF8_TO_TCHAR(OpList->ops[i].disconnect.reason));
 			AsyncTask(ENamedThreads::GameThread, [this, ErrorMessage]
 			{
-				GetSpatialNetDriverChecked()->HandleOnConnectionFailed(ErrorMessage);
+				UGameInstance* GameInstance = Cast<UGameInstance>(GetOuter());
+				if (GEngine != nullptr && GameInstance->GetWorld() != nullptr)
+				{
+					GEngine->BroadcastNetworkFailure(GameInstance->GetWorld(), GetSpatialNetDriverChecked(), ENetworkFailure::PendingConnectionFailure, *ErrorMessage);
```

Comment:
Why does this not translate the error message received? If we have arranged it such that this always give `PendingConnectionFailure` then we don't really get anything out of parsing the timeout differently from the others when disconnecting a healthy worker.